Crucible.GameAPI/CrucibleNpcAppearance.cs

```diff
@@ -547,6 +547,7 @@ private T RequirePart<T>()
             var part = this.npcTemplate.baseParts.OfType<T>().FirstOrDefault();
             if (part == null)
             {
+                // TODO Depending on the intended use case of this method npcTemplate.closenessParts may also need be checked.
                 throw new InvalidOperationException($"NPC template {this.npcTemplate.name} does not have a {typeof(T).Name} part.");
```

Comment:
Expanding on this comment: It looks like this method is only used for specific parts which were not moved to the closenessParts list. However this method could have unexpected results in the future if it is expected to return parts within the closenessParts list.